R/gen_OpenSpecy.R

```diff
@@ -0,0 +1,85 @@
+#' @title Generic Open Specy Functions
+#' @rdname gen_OpenSpecy
+#'
+#' @description
+#' Functions to manipulate and visualize OpenSpecy objects.
+#'
+#' @details
+#' \code{as_OpenSpecy()} creates an OpenSpecy object.
+#' \code{print.OpenSpecy()} prints the contents of an OpenSpecy object.
+#' \code{head.OpenSpecy()} shows the first few lines of an OpenSpecy object.
+#' \code{sample.OpenSpecy()} samples spectra from an OpenSpecy object.
+#'
+#' @param x an OpenSpecy object.
+#' @param \ldots args.
+#' @param size a positive integer specifying the number of items to select from the spectra. (only for \code{sample.OpenSpecy()})
+#' @param replace should sampling be done with replacement? (only for \code{sample.OpenSpecy()})
+#' @param prob a vector of probability weights for obtaining the elements of the spectra being sampled. (only for \code{sample.OpenSpecy()})
+#'
+#' @return
+#' \code{print.OpenSpecy()} returns a textual representation of an OpenSpecy object.
+#' \code{head.OpenSpecy()} returns the first few lines of an OpenSpecy object.
+#' \code{sample.OpenSpecy()} returns an OpenSpecy object with a subset of the spectra.
+#'
+#' @examples
+#'
+#' # Creating an OpenSpecy object
+#' sample_OpenSpecy <- read_any(read_extdata("raman_hdpe.csv"))
+#'
+#' # Printing the OpenSpecy object
+#' print.OpenSpecy(sample_OpenSpecy)
+#'
+#' # Displaying the first few lines of the OpenSpecy object
+#' head(sample_OpenSpecy)
+#'
+#'
+#' # Sampling the OpenSpecy object
+#' sample_OpenSpecy <- read_any(read_extdata("CA_tiny_map.zip"))
+#' sampled_OpenSpecy <- sample.OpenSpecy(sample_OpenSpecy, size = 5)
+#' print.OpenSpecy(sampled_OpenSpecy)
+#'
+#' @author
+#' Zacharias Steinmetz, Win Cowger
+#'
+#' @seealso
+#' Other related functions.
+#'
+#' @importFrom magrittr %>%
+#' @importFrom utils head
+#' @importFrom tidyr pivot_longer
+#' @importFrom ggplot2 ggplot aes geom_line theme_minimal labs
```

Comment:
Why is ggplot2 imported here?

Reply:
It isn't needed. At one point, I had the plots here and just forgot to remove it.